tests/utapi/awsNodeSdk.js

```diff
@@ -1,159 +1,194 @@
+                // Create a new hash instance for each request
+                headers['Content-MD5'] = createHash('md5').update(bodyContent).digest('base64');
+            }
+
+            return await next(args);
+        },
+        {
+            step: 'build',
+        }
+    );
+    return client;
+}
+
 function wait(timeoutMs, cb) {
     setTimeout(cb, timeoutMs);
 }
+
 function createBucket(bucket, cb) {
-    return s3Client.createBucket({
-        Bucket: bucket,
-    }, (err, data) => {
-        assert.ifError(err);
-        return cb(err, data);
-    });
+    return s3Client.send(new CreateBucketCommand({ Bucket: bucket }))
+        .then(data => cb(null, data))
+        .catch(cb);
 }
+
 function deleteBucket(bucket, cb) {
-    return s3Client.deleteBucket({
-        Bucket: bucket,
-    }, err => {
-        assert.ifError(err);
-        return cb(err);
-    });
+    return s3Client.send(new DeleteBucketCommand({ Bucket: bucket }))
+        .then(() => cb())
+        .catch(cb);
 }
+
 function putObject(bucket, key, size, cb) {
-    return s3Client.putObject({
+    const body = Buffer.alloc(size);
+    const params = {
         Bucket: bucket,
         Key: key,
-        Body: Buffer.alloc(size),
-    }, (err, data) => {
-        assert.ifError(err);
-        return cb(err, data);
-    });
+        Body: body,
+    };
+    return s3Client.send(new PutObjectCommand(params))
+        .then(data => cb(null, data))
+        .catch(cb);
 }
+
 function deleteObject(bucket, key, cb) {
-    return s3Client.deleteObject({
-        Bucket: bucket,
-        Key: key,
-    }, err => {
-        assert.ifError(err);
-        return cb(err);
-    });
+    return s3Client.send(new DeleteObjectCommand({ Bucket: bucket, Key: key }))
+        .then(() => cb())
+        .catch(cb);
 }
+
 function deleteObjects(bucket, keys, cb) {
-    const objects = keys.map(key => {
-        const keyObj = {
-            Key: key,
-        };
-        return keyObj;
-    });
+    const objects = keys.map(key => ({ Key: key }));
+    const deleteRequest = { Objects: objects, Quiet: true };
     const params = {
         Bucket: bucket,
-        Delete: {
-            Objects: objects,
-            Quiet: true,
-        },
+        Delete: deleteRequest,
     };
-    return s3Client.deleteObjects(params, err => {
-        assert.ifError(err);
-        return cb(err);
-    });
+    return s3Client.send(new DeleteObjectsCommand(params))
+        .then(() => cb())
+        .catch(cb);
 }
+
 function copyObject(bucket, key, cb) {
-    return s3Client.copyObject({
-        Bucket: bucket,
-        CopySource: `/${bucket}/${key}`,
-        Key: `${key}-copy`,
-    }, err => {
-        assert.ifError(err);
-        return cb(err);
-    });
+    const params = { Bucket: bucket, CopySource: `${bucket}/${key}`, Key: `${key}-copy` };
+    return s3Client.send(new CopyObjectCommand(params))
+        .then(() => cb())
```

Comment:
```suggestion
        .then(cb)
```

Same for other iterations